File: client-app/api/note.ts
import api from "./index";

// Note 인터페이스 정의
export interface Note {
    noteId: number;
    title: string;
    userId: number;
    content: string;
    password: string;
    postDate: string;
    modifyDate?: string;
    thumbImage?: string;
    category: string;
    subCategory?: string;
    readCount?: number;
    isPost: string;
    postIp: string;
    modifyIp?: string;
}

// PostedNote 타입 정의
export type PostedNote = Omit<Note, "password">;

// PostNote 인터페이스 정의
export interface PostNote {
    noteId?: number;
    title: string;
    userId: number;
    content: string;
    thumbImage?: string;
    category: string;
    subCategory?: string;
    isPost?: string;
    password?: string;
}

// NoteSummary 타입 정의
export type NoteSummary = {
    noteId: number;
    title: string;
}

// CategoryView 타입 정의
export type CategoryView = {
    userId: number;
    category: string;
    subCategory: string;
}

// SidebarCategoryView 타입 정의
export type SidebarCategoryView = {
    name: string;
    mainCount: number;
    subName: string;
    subCount: number;
}

// IpLogModel 타입 정의
export type IpLogModel = {
    visitorIp: string;
    blogId: number;
}

// Note API 함수 정의
const noteApi = {
    postNote: (formType: number, payload: PostNote) =>
        api.post<number>(`/api/notes?formType=${formType}`, payload),

    postCategory: (category: CategoryView) =>
        api.post(`/api/notes/categories`, category),

    postIpLog: (payload: IpLogModel) =>
        api.post(`/api/notes/ipLog`, payload),

    getNoteAll: (userId: number) =>
        api.get<PostedNote[]>(`/api/notes/userId=${userId}`),

    getNoteSummary: () => 
        api.get<NoteSummary[]>(`/api/notes/summary`),

    getNoteById: (id: number) =>
        api.get<PostedNote>(`/api/notes/detail?id=${id}`),

    getNoteByCategory: (id: number, category: string, subCategory: string) =>
        api.get<PostedNote[]>(`/api/notes/category?id=${id}&category=${category}&subCategory=${subCategory}`),

    getNoteBySearch: (content: string) =>
        api.get<PostedNote[]>(`/api/notes/search?query=${content}`),

    deleteNoteById: (id: number) =>
        api.delete(`/api/notes/${id}`),

    getCategoryList: () =>
        api.get<CategoryView[]>(`/api/notes/categories`),

    getSidebarCategoryList: () =>
        api.get<SidebarCategoryView[]>(`/api/notes/sidebar/categories`),


    getTotalReadCount: () =>
        api.get<number>(`/api/notes/totalReadCount`),

    getTodayReadCount: () =>
        api.get<number>(`/api/notes/todayReadCount`),

    saveImage: (file: FormData) =>
        api.post(`/api/notes/saveImage`, file, {
            headers: {
                'Content-Type': 'multipart/form-data'
            }
        })
};

export default noteApi;

File: Model/BlogNote/Note.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace next_core_blog.Model.BlogNote
{
    public class Note
    {
        [Display(Name = "번호")]
        public int noteId { get; set; }
        [Display(Name = "제목")]
        [Required(ErrorMessage = "* 제목을 작성해 주세요")]
        public string title { get; set; }
        [Display(Name = "유저 번호")]
        public int userId { get; set; }
        [Display(Name = "내용")]
        [Required(ErrorMessage = "* 내용을 작성해 주세요")]
        public string content { get; set; }
        [Display(Name = "비밀번호")]
        [Required(ErrorMessage = "* 비밀번호를 입력해 주세요")]
        public string password { get; set; }
        public string postDate { get; set; }
        public string modifyDate { get; set; }
        public string thumbImage { get; set; }
        public string category { get; set; }
        public string subCategory { get; set; }
        public int readCount { get; set; }
        public char isPost { get; set; }
        public string postIp { get; set; }
        public string modifyIp { get; set; }

    }
}